### PR TITLE
User/provencher/exposed box activation setting

### PR DESCRIPTION
**Pull request by provencher**
Overview
---
Previously, the box activation mode was set to Manual, which meant that changing handle visibility settings would not be persisted as the pointer focus change listener would prevent handle activation depending on the box activation mode.

Changes
---
- Fixes: # https://github.com/Microsoft/MixedRealityToolkit-Unity/issues/4052
-> The above issue is resolved when setting the activation type to ActivateByProximityAndPointer

### CODE REVIEW

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs [hunk 1]
```diff
@@ -87,7 +87,7 @@ private enum BoundsCalculationMethod
             Renderers,
             MeshFilters
         }
-        private enum BoundingBoxActivationType
+        public enum BoundingBoxActivationType
```
Comment (Jarodshow) [inline Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: Since this is public now shouldn't it get its own file?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: Agreed

Comment (provencher) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: Can we just expose all these enums then?

I've exposed what I want, but I'd like anything serializable to be exposed.

The issue with making this it's own file as well is that it's only used in BoundingBox. Is there really a need to split that from the source file?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: Mainly I don't want to have the hassle of writing the `Class.EnumType` every time. 
But it's also a legibility issue and part of the coding guidelines.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: If it were possible to make the enum a bit more generic, that'd be great too

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: This enum _does_ seem rather useful in more than just bounding box activations

Comment (provencher) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: Well a few thoughts on that.
`Class.EnumType` helps with consistency regarding where the enum type is used. That's actually a pro to me, if it's only used in one spot.

Regarding the re-usability of the enum however, I think we could move it out once it's needed elsewhere. As in, if someone PR's a change that makes use of it elsewhere, then it should be moved to another time. Until then, I think the way it currently is setup is sufficient.

**Comment by wiwei** [comment]
I want to give @luis-valverde-ms a chance to look at this.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs (hunk 1 above)
Comment (luis-valverde-ms) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: > Regarding the re-usability of the enum however, I think we could move it out once it's needed elsewhere. As in, if someone PR's a change that makes use of it elsewhere, then it should be moved to another time. Until then, I think the way it currently is setup is sufficient.

+1

**Comment by luis-valverde-ms** [comment]
The intent of the changes looks reasonable to me but the component owners (@cre8ivepark and @johnppella?) should have a look too.

The changes seem to deal with being able to change the activation type at runtime (from code) but the PR description doesn't make it clear. Can we rework that a bit and make it clear too how it solves the connected issue?

**Comment by provencher** [comment]
@luis-valverde-ms this is from the referenced issue 

"So the jist is that I was setting which handles the bounding box would show, as your code example pointed out, but when it came time to hover over the box, no handles would show up at all.

I looked through the code, and this is because the focus change function in the bounding box returns before changing handle visibility if the enum I exposed was set to manual, which is the default value. This early return prevents the handles that should be visible from showing, as the code that determines visibility does a check against the show wireframe property, which is set on the focus change function.

Perhaps this exposes a deeper bug, but I didn't want to change core functionality of the box as that could have far reaching consequences. Simply changing the enum I exposed stops the focus change function from returning early, and the wireframe bool is properly set.

Part of the issue with my bug report is that I conflated manipulation constraints with bounding box handles thinking they were more related.

Perhaps constraints on the manipulation handler should affect which handles the bounding box shows, as the handles permit transformations the manipulation handler might forbid."

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs (hunk 1 above)
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: No worries, I understand :)

Just wanted to keep the project consistent and in line with the established patterns and in line with the coding guidelines. Just have to speak up and have it noted for the record.

imho it's better to make this change now than to have a breaking change later when we decide to make that kinda change.

Comment (provencher) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs]: Yeah that makes sense. I didn't consider it breaking code down the line. It's a tradeoff

**Comment by wiwei** [comment]
I'll followup with @cre8ivepark to see if he can look at this.

**Pull request merged**